recipes/gdal/post_3.5.0/patches/3.5.1/0-replace-find-package.patch

```diff
@@ -0,0 +1,264 @@
+diff --git a/alg/CMakeLists.txt b/alg/CMakeLists.txt
+index edf75158c7..4200309ca8 100644
+--- a/alg/CMakeLists.txt
++++ b/alg/CMakeLists.txt
+@@ -72,7 +72,7 @@ if (GDAL_USE_OPENCL)
+   target_sources(alg PRIVATE gdalwarpkernel_opencl.h gdalwarpkernel_opencl.cpp)
+ endif ()
+ 
+-gdal_target_link_libraries(alg PRIVATE PROJ::proj)
++target_link_libraries(alg PUBLIC PROJ::proj)
+ 
+ if (GDAL_USE_QHULL_INTERNAL)
+   target_compile_definitions(alg PRIVATE -DINTERNAL_QHULL)
+diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
+index a49165d14a..91d6170067 100644
+--- a/apps/CMakeLists.txt
++++ b/apps/CMakeLists.txt
+@@ -25,7 +25,7 @@ target_include_directories(
+   appslib PRIVATE $<TARGET_PROPERTY:gdal_vrt,SOURCE_DIR> $<TARGET_PROPERTY:ogrsf_generic,SOURCE_DIR>
+                   $<TARGET_PROPERTY:ogr_geojson,SOURCE_DIR> $<TARGET_PROPERTY:ogr_MEM,SOURCE_DIR>)
+ 
+-gdal_target_link_libraries(appslib PRIVATE PROJ::proj)
++target_link_libraries(appslib PUBLIC PROJ::proj)
+ 
+ set_property(TARGET appslib PROPERTY POSITION_INDEPENDENT_CODE ${GDAL_OBJECT_LIBRARIES_POSITION_INDEPENDENT_CODE})
+ if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.16)
+diff --git a/cmake/helpers/CheckDependentLibraries.cmake b/cmake/helpers/CheckDependentLibraries.cmake
+index 7fa3b565c7..8df61c5293 100644
+--- a/cmake/helpers/CheckDependentLibraries.cmake
++++ b/cmake/helpers/CheckDependentLibraries.cmake
+@@ -11,7 +11,7 @@ Detect GDAL dependencies and set variable HAVE_*
+ include(CheckFunctionExists)
+ include(CMakeDependentOption)
+ include(FeatureSummary)
+-include(DefineFindPackage2)
++include(ConanFindPackage)
+ include(CheckSymbolExists)
+ 
+ option(
+@@ -110,49 +110,6 @@ macro (gdal_check_package name purpose)
+   set(_find_dependency "")
+   set(_find_dependency_args "")
+   find_package2(${name} QUIET OUT_DEPENDENCY _find_dependency)
+-  if (NOT DEFINED ${key}_FOUND)
```

Comment:
Could this section be replace with a `message(FATAL_ERROR ...)` instead.. you be smaller and easier to understand with a good message like `"Conan recipes should rely on config files from generators so let's just disble GDAL's"` something like that

Reply:
Good idea, done.